competition/sweep/SumoDistanceSensor.ino

```diff
@@ -0,0 +1,486 @@
+  //motors.flipRightMotor(true);
+
+  pinMode(LED, HIGH);
+  buzzer.playMode(PLAY_AUTOMATIC);
+  waitForButtonAndCountDown(false);
+}
+
+void waitForButtonAndCountDown(bool restarting)
+{
+#ifdef LOG_SERIAL
+  Serial.print(restarting ? "Restarting Countdown" : "Starting Countdown");
+  Serial.println();
+#endif
+
+  digitalWrite(LED, HIGH);
+  button.waitForButton();
+  digitalWrite(LED, LOW);
+
+  // play audible countdown
+  for (int i = 0; i < 3; i++)
+  {
+    delay(1000);
+    buzzer.playNote(NOTE_G(3), 50, 12);
+  }
+  delay(1000);
+  buzzer.playFromProgramSpace(sound_effect);
+  delay(1000);
+
+  // reset loop variables
+  in_contact = false;  // 1 if contact made; 0 if no contact or contact lost
+  contact_made_time = 0;
+  last_turn_time = millis();  // prevents false contact detection on initial acceleration
+  _forwardSpeed = SearchSpeed;
+  full_speed_start_time = 0;
+}
+
+void loop()
+{
+  if (button.isPressed())
+  {
+    // if button is pressed, stop and wait for another press to go again
+    motors.setSpeeds(0, 0);
+    button.waitForRelease();
+    waitForButtonAndCountDown(true);
+  }
+
+  loop_start_time = millis();
+  lsm303.readAcceleration(loop_start_time);
+  sensors.read(sensor_values);
+
+  if ((_forwardSpeed == FullSpeed) && (loop_start_time - full_speed_start_time > FULL_SPEED_DURATION_LIMIT))
+  {
+    setForwardSpeed(SustainedSpeed);
+  }
+
+  if (sensor_values[0] < QTR_THRESHOLD)
+  {
+    // if leftmost sensor detects line, reverse and turn to the right
+    turn(RIGHT, true);
+  }
+  else if (sensor_values[5] < QTR_THRESHOLD)
+  {
+    // if rightmost sensor detects line, reverse and turn to the left
+    turn(LEFT, true);
+  }
+  else  // otherwise, go straight
+  {
+    if (check_for_contact()) on_contact_made();
+    int speed = getForwardSpeed();
+    motors.setSpeeds(speed, speed);
+  }
+}
+
+// execute turn
+// direction:  RIGHT or LEFT
+// randomize: to improve searching
+void turn(char direction, bool randomize)
+{
+#ifdef LOG_SERIAL
+  Serial.print("turning ...");
+  Serial.println();
+#endif
+
+  // assume contact lost
+  on_contact_lost();
+
+  static unsigned int duration_increment = TURN_DURATION / 4;
+
+  // motors.setSpeeds(0,0);
+  // delay(STOP_DURATION);
+  motors.setSpeeds(-REVERSE_SPEED, -REVERSE_SPEED);
+  delay(REVERSE_DURATION);
+  motors.setSpeeds(TURN_SPEED * direction, -TURN_SPEED * direction);
+  delay(randomize ? TURN_DURATION + (random(8) - 2) * duration_increment : TURN_DURATION);
+  int speed = getForwardSpeed();
+  motors.setSpeeds(speed, speed);
+  last_turn_time = millis();
+}
+
+// execute turn to specific angle (approximately)
+// angle: -180 to 180 degrees, specifying what angle to turn to
+void turnToAngle(int angle)
+{
+  int direction = (angle < 0) ? LEFT : RIGHT;
+
+  motors.setSpeeds(TURN_SPEED * direction, -TURN_SPEED * direction);
+  unsigned int duration = (abs(angle) * HALF_TURN_DURATION) / 180;
+  delay(duration);
+  motors.setSpeeds(0, 0);
+}
```

Comment:
This is the new function I added - we'd still need to calibrate the values to ensure it turns to the correct angle